doc/computing/randomness.rst

```diff
@@ -0,0 +1,283 @@
+.. Places parent toc into the sidebar
+
+:parenttoc: True
+
+.. _randomness:
+
+Controlling randomness: good practices and common pitfalls
+==========================================================
+
+Some scikit-learn objects are inherently random. These are usually estimators
+(e.g. :class:`~sklearn.ensemble.RandomForestClassifier`) and cross-validation
+splitters (e.g. :class:`~sklearn.model_selection.KFold`). The randomness of
+these objects is controlled via their `random_state` parameter, as described
+in the :term:`Glossary <random_state>`. This section expands on the glossary
+entry, and describes good practice and common pitfalls w.r.t. to this
+subtle parameter.
+
+The `random_state` parameter determines whether multiple calls to :term:`fit`
+(for estimators) or to :term:`split` (for cv spliters) will produce the same
+results, according to these rules:
+
+- If an int is passed, calling `fit()` or `split()` multiple times yields the
+  same results.
+- If `None` or a `RandomState` instance is passed: `fit()` and `split()` will
+  yield different results each time they are called, producing an event chain
+  of maximal entropy. `None` is the default value for all `random_state`
+  parameters.
+
+  .. note::
+      Since passing `random_state=None` is equivalent to passing the global
+      `RandomState` instance from `numpy`
+      (`random_state=np.random.mtrand._rand`), we will not explicitly mention
+      `None` here, but everything that applies to instances also applies to
+      using `None`.
+
+.. warning:: TLDR
+
+    Unless you know what you are doing, we strongly recommend to use integers
+    as the `random_state` parameter of estimators and cv splitters. Leaving
+    the default (`None`) or using `RandomState` instances is sometimes
+    useful, but can have surprising effects.
```

Comment:
```suggestion
    By passing integers as the `random_state` parameter of estimators and cv splitters
    makes your code more reproducible and allows you to isolate sources of variance
    in your analysis. It can therefore be recommended and be seen as a good practice.
    This however can hide random effects that are desirable to assess statistical
    significance of your results. This explains why the default (`None`) or using
    `RandomState` instances is sometimes useful, but can have surprising effects.
```